qir/qat/Version/version_builder.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import sys
+from string import Template
+
+version_file, template, output = sys.argv[1:]
+with open(version_file, "r") as fb:
+    values = [x.strip().split(" ", 1) for x in fb.readlines() if " " in x.strip()]
+    kwargs = dict(values)
```

Comment:
What if the command line args (`sys.argv[1:]`) are not provided?
I assume that 
* `version_file` is checked against being initialized before calling ` open(version_file, ...`;
* the file is accessed (`fb.readlines()`) only if the open succeeded (`open(version_file, "r") as fb`).

Reply:
If `sys.argv[1:]` is not provided, the program fails with an exception.